src/tests/epics/epicSelection.ts

```diff
@@ -183,68 +184,91 @@ export interface Result {
 export const findTestAndVariant = (
     tests: EpicTest[],
     targeting: EpicTargeting,
+    superModeArticles: SuperModeArticle[],
     epicType: EpicType,
     includeDebug: boolean = false,
 ): Result => {
     const debug: Debug = {};
 
     const userCohorts = getUserCohorts(targeting);
 
-    // Also need to include canRun of individual variants (only relevant for
-    // manually configured tests).
-    // https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js#L378
-    const filters: Filter[] = [
-        shouldNotRender(epicType),
-        isOn,
-        isNotExpired(),
-        hasSectionOrTags,
-        userInTest(targeting.mvtId || 1),
-        inCorrectCohort(userCohorts),
-        excludeSection,
-        excludeTags,
-        hasCountryCode,
-        matchesCountryGroups,
-        withinMaxViews(targeting.epicViewLog || []),
-        withinArticleViewedSettings(targeting.weeklyArticleHistory || []),
-        respectArticleCountOptOut,
-    ];
+    const getFilters = (isSuperModePass: boolean): Filter[] => {
+        return [
+            shouldNotRender(epicType),
+            isOn,
+            isNotExpired(),
+            hasSectionOrTags,
+            userInTest(targeting.mvtId || 1),
+            inCorrectCohort(userCohorts),
+            excludeSection,
+            excludeTags,
+            hasCountryCode,
+            matchesCountryGroups,
+            // For the super mode pass, we treat all tests as "always ask" so disable this filter
+            ...(isSuperModePass ? [] : [withinMaxViews(targeting.epicViewLog || [])]),
+            withinArticleViewedSettings(targeting.weeklyArticleHistory || []),
+            respectArticleCountOptOut,
+        ];
+    };
+
+    const filterTests = (tests: EpicTest[], filters: Filter[]): EpicTest | undefined => {
+        const test = tests.find(test =>
+            filters.every(filter => {
+                const got = filter.test(test, targeting);
+
+                if (debug[test.name]) {
+                    (debug[test.name] as FilterResults)[filter.id] = got;
+                } else {
+                    debug[test.name] = { [filter.id]: got };
+                }
+
+                if (!got && process.env.LOG_FAILED_TEST_FILTER === 'true') {
+                    console.log(`filter failed: ${test.name}; ${filter.id}`);
+                }
+
+                return got;
+            }),
+        );
+
+        return test;
+    };
+
+    const filterTestsWithSuperModePass = (tests: EpicTest[]): EpicTest | undefined => {
+        const test = filterTests(tests, getFilters(false));
+        if (test) {
+            return test;
+        }
+
+        const superModeTest = filterTests(tests, getFilters(true));
+        if (superModeTest) {
+            return superModeify(superModeTest);
+        }
+
+        return undefined;
```

Comment:
This could also be written as:

```ts
return (
  filterTests(tests, getFilters(false)) ?? 
  superModeify(filterTests(test, getFilters(true))
)
```

but perhaps that is a little too dense?

Reply:
I like the short version!

Reply:
nice! I'll update it - it feels a lot more functional 👍